test/test_spark.py

```diff
@@ -22,17 +22,25 @@
 import pytest
 import re
 import subprocess
+import sys
 import time
 import torch
 import unittest
 import warnings
 
+from horovod.run.common.util import secret
 from horovod.run.mpi_run import _get_mpi_implementation_flags
 import horovod.spark
+from horovod.spark.task.task_service import SparkTaskService, SparkTaskClient
 import horovod.torch as hvd
 
 from mock import MagicMock
 
+if sys.version_info[0] < 3:
```

Comment:
Rather than add a dependency on `backports.tempfile`, I think it would be easier to just write our own temp directory function.  We already did this in an upcoming PR (https://github.com/horovod/horovod/blob/spark_estimator/test/common.py#L63).  Maybe you can cherry pick that context manager and use it here?

Reply:
I know, this import and `backports.tempfile` dependency is ugly, but I'd prefer that over duplicating code, even if it is small.

A related question that may help me assess the situation: how long do you plan to support Python2.7? It will die in a month. Is it worth adding this dependency / your own implementation when this is only needed to support Python2.7, which will go away short term anyway?

Reply:
Most likely we will drop Python 2.7 support at the end of the year, but that decision needs to be brought before the steering committee, in case anyone is stuck with Python 2.7 for whatever reason.

If you're uncomfortable adding that extra function, one option would be to simply skip this test for Python 2 by adding the following decorator:

```
@pytest.mark.skipif(sys.version_info < (3, 0), reason="tempfile.TemporaryDirectory only available in Python 3")
```

Reply:
Interesting alternative.

Since you are introducing the `TemporaryDirectory` anyway, I will use it. But it should get removed once Python2.7 support is dropped.

Reply:
@tgaddair is this okay to merge now?  It would be really useful to have in the next Horovod release

Reply:
Sorry for the delay!  LGTM!